docs/src/components/Utils/ApiTable.component.tsx

```diff
@@ -0,0 +1,161 @@
+    in situations where a combination of components and custom text is needed as ReactHTMLParser
+    will end up rendering passed components as [Object, object].
+
+    Ex:
+
+    <ApiTable data={[
+        {
+            property: 'background',
+            description: [
+                <ColorBlurb label="DropDown" background gradient />,
+                ` Note that changing this value changes both the arrow color and the background
+                color of the <pre>DropDown</pre>, so using anything other than solid colors can have
+                unexpected results.`
+            ],
+            type: 'string',
+            default: 'light',
+        },
+        {
+            property: 'color',
+            description: <ColorBlurb label="Example Component" />,
+            type: 'string',
+            default: 'black',
+        },
+        {
+            property: 'title',
+            description: 'Title for this <pre>Component</pre>.',
+            type: 'string',
+        },
+    ]} />
+
+    TODO: Eventually remove the semantic table option from the site and the styles associated with
+    it.
+*/
+
+// REACT
+import * as React from 'react';
+// VENDOR
+import styled from '@xstyled/styled-components';
+import { th } from '@xstyled/system';
+import parse from 'html-react-parser';
+// ANCHOR & COMPONENTS
+import { colors } from '@retailmenot/anchor';
+import { InlineCodeStyle } from '../Layout/Page/Page.component';
+
+const Table = styled('table')`
+    width: 100%;
+    font-family: ${th('typography.fontFamily')};
+    margin: 1rem 0;
+
+    th {
+        text-align: left;
+        padding: 0.75rem 0.5rem;
+        border-bottom: solid thin ${colors.ash.light};
+        background-color: ${colors.white.base};
+    }
+    tr {
+        &:nth-child(odd) {
+            background-color: ${colors.silver.light};
+        }
+    }
+    td {
+        padding: 0.75rem 0.5rem;
+        line-height: 1.5rem;
+
+        &:first-child {
+            font-family: monospace;
+            color: ${colors.flashPink.dark};
+            width: 10%;
+        }
+        &:nth-child(2) {
+            width: 60%;
+        }
+        &:nth-child(3) {
+            font-family: monospace;
+            color: ${colors.cyberMango.dark};
+            width: 20%;
+        }
+        &:nth-child(4) {
+            text-align: left;
+            font-family: monospace;
+            width: 10%;
+            white-space: nowrap;
+        }
+
+        pre {
+            ${InlineCodeStyle};
+        }
+    }
+`;
+
+interface ApiTableProps {
+    data: [];
+}
+
+interface RowData {
+    property: string;
+    /* There are several utility components that could be passed as a description in addition to a
+    plain string. Also an option is to pass an array of values in order to combine both plain text
+    and components.  */
+    description: string | [] | React.ReactElement;
```

Comment:
Eventually I think we should have some default `descriptions` based on the `property`. For the kind of props that are basically the same everywhere (e.g. `borderRadius`), to reduce the copy pasta.

Looks good tho

Reply:
Absolutely agree. I started doing that with the `ColorBlurb` component and intend on expanding on that trend.